src/main/java/uk/gov/hmcts/darts/retention/service/impl/ApplyRetentionCaseAssociatedObjectsSingleCaseProcessorImpl.java

```diff
@@ -73,14 +76,20 @@ public void processApplyRetentionToCaseAssociatedObjects(Integer caseId) {
     }
 
     private void applyRetentionToMedias(CourtCaseEntity courtCase) {
-
-        var medias = mediaRepository.findAllByCaseId(courtCase.getId());
-
-        for (var media : medias) {
-            var cases = media.associatedCourtCases();
-            if (allClosed(cases)) {
-                setLongestRetentionDateForMedia(media, cases);
-                setRetentionConfidenceScoreAndReasonForMedia(media, cases);
+        List<MediaEntity> mediaEntities = new ArrayList<>();
+        // get all current media linked to the case
+        mediaEntities.addAll(mediaRepository.findAllByCaseId(courtCase.getId()));
+        // get all media linked to the case via media linked case
+        mediaEntities.addAll(mediaRepository.findAllLinkedByMediaLinkedCaseByCaseId(courtCase.getId()));
+        // remove duplicates
+        var allMedia = io.vavr.collection.List.ofAll(mediaEntities).distinctBy(MediaEntity::getId).toJavaList();
```

Comment:
Optional: but could use java streams as they tend to be more performant.
mediaEntities.stream().distinct().toList()

Reply:
This was done using the id as there are 2 calls to get the media entities and as they have a last modified date they could be different so its better to just compare against the media id not the whole object